File: 2024 May Leetcoding Challenge/3075. Maximize Happiness of Selected Children.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        int n = happiness.length;
        Arrays.sort(happiness);
        long count = 0, cnt = 0;
        for(int i = n - 1; i >= 0 && k > 0; i--){
            k--;
            
            if(happiness[i] - cnt > 0)
                count += happiness[i] - cnt;
            cnt++;
        }
        return count;
    }
}

File: 2025 December Leetcoding Challenge/3075. Maximize Happiness of Selected Children.cpp
class Solution {
public:
    long long maximumHappinessSum(vector<int>& happiness, int k) {
        int n = happiness.size();
        sort(happiness.rbegin(), happiness.rend());
        long long sum = 0;
        for(int i = 0; i < k && i < n; ++i){
            sum += max(0, happiness[i] - i);
        }
        return sum;
    }
};
